source/shadow.c

```diff
@@ -527,41 +710,59 @@ ShadowStatus_t Shadow_MatchTopic( const char * pTopic,
         {
             *pThingNameLength = thingNameLength;
         }
+
+        if( pShadowName != NULL )
+        {
+            *pShadowName = &( pTopic[ SHADOW_PREFIX_LENGTH + thingNameLength +
```

Comment:
If the shadowNameLength is zero we want to return NULL for this variable.

Reply:
I agree with your point, but we can't reach this line of code with shadowNameLength of 0. The validation of the shadow name will have already failed. The function will return SHADOW_SHADOWNAME_PARSE_FAILED. There is a unit test case for it using topic TEST_NAMED_TOPIC_STRING_INVALID_SHADOWNAME.

Reply:
Per discussions, changed to use Shadow_GetTopicString() (instead of the macros) so we could do the subroutine solution.